src/test/regress/expected/failure_ddl.out

```diff
@@ -0,0 +1,1228 @@
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT count(*) FROM public.table_attrs where relid = 'test_table'::regclass;
+ count 
+-------
+     2
+(1 row)
+
+-- cancel as soon as the coordinator sends begin
+SELECT citus.mitmproxy('conn.onQuery(query="^BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED").cancel(' ||  pg_backend_pid() || ')');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+ALTER TABLE test_table ADD COLUMN new_column INT;
+ERROR:  canceling statement due to user request
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT count(*) FROM public.table_attrs where relid = 'test_table'::regclass;
+ count 
+-------
+     2
+(1 row)
+
+-- kill as soon as the coordinator sends worker_apply_shard_ddl_command
+SELECT citus.mitmproxy('conn.onQuery(query="worker_apply_shard_ddl_command").kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+ALTER TABLE test_table ADD COLUMN new_column INT;
+ERROR:  server closed the connection unexpectedly
+	This probably means the server terminated abnormally
+	before or while processing the request.
+CONTEXT:  while executing command on localhost:57640
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT count(*) FROM public.table_attrs where relid = 'test_table'::regclass;
+ count 
+-------
+     2
+(1 row)
+
+-- cancel as soon as the coordinator sends worker_apply_shard_ddl_command
+SELECT citus.mitmproxy('conn.onQuery(query="worker_apply_shard_ddl_command").cancel(' ||  pg_backend_pid() || ')');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+ALTER TABLE test_table ADD COLUMN new_column INT;
+ERROR:  canceling statement due to user request
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT count(*) FROM public.table_attrs where relid = 'test_table'::regclass;
+ count 
+-------
+     2
+(1 row)
+
+-- killing on PREPARE should be fine, everything should be rollbacked
+SELECT citus.mitmproxy('conn.onCommandComplete(command="PREPARE TRANSACTION").kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+ALTER TABLE test_table ADD COLUMN new_column INT;
+ERROR:  connection not open
+CONTEXT:  while executing command on localhost:57640
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT count(*) FROM public.table_attrs where relid = 'test_table'::regclass;
+ count 
+-------
+     2
+(1 row)
+
+SELECT run_command_on_placements('test_table', $$SELECT count(*) FROM public.table_attrs where relid = '%s'::regclass$$) ORDER BY 1;
+  run_command_on_placements   
+------------------------------
+ (localhost,57637,100804,t,2)
+ (localhost,57637,100805,t,2)
+ (localhost,57637,100806,t,2)
+ (localhost,57637,100807,t,2)
+ (localhost,57640,100804,t,2)
+ (localhost,57640,100805,t,2)
+ (localhost,57640,100806,t,2)
+ (localhost,57640,100807,t,2)
+(8 rows)
+
+-- we should be able to revocer the transaction and
+-- see that the command is rollbacked
```

Comment:
see that what command is rollbacked? Above all shards have two columns, and below they also have two columns.